reframe/core/pipeline.py

```diff
@@ -676,10 +681,10 @@ def supports_system(self, partition_name):
         return partition_name in self.valid_systems
 
     def supports_environ(self, env_name):
-        if '*' in self.valid_prog_environs:
-            return True
-
-        return env_name in self.valid_prog_environs
+        for env in self.valid_prog_environs:
+            if fnmatch.fnmatch(env_name, env):
```

Comment:
I like this and it's simple. Would it make sense to support wildcards also in the `supports_system()`?

Reply:
It is possible to do so if you want to support also wildcards for systems.